cache_test.go

```diff
@@ -131,6 +131,29 @@ func TestOverwrite(t *testing.T) {
 
 }
 
+func TestGetWithExpiration(t *testing.T) {
+	cache := NewCache(1024)
+	key := []byte("abcd")
+	val := []byte("efgh")
+	err := cache.Set(key, val, 2)
+	if err != nil {
+		t.Error("err should be nil", err.Error())
+	}
+
+	res, expiry, err := cache.GetWithExpiration(key)
+	time.Sleep(time.Second)
+	if err != nil {
+		t.Error("err should be nil", err.Error())
+	}
+	if !bytes.Equal(val, res) {
+		t.Fatalf("%s should be the same as %s but isn't", res, val)
+	}
+	now := time.Now()
+	if expiry != uint32(now.Unix()+1) {
```

Comment:
I mean the unstable factor is that we may sleep more than one second.
Say if we set the value at `11:11:11.999999`, after the sleep, the time may be `11:11:13.000001`. Then the check fails.

Reply:
Ok, I made the test way less brittle.
Thanks to @dkowis ;-)